### ISSUE TITLE
PremiumTo "Down" (Downloading Empty Files)

### ISSUE THREAD

**Issue by sasese** [pyload/pyload#1337]
After months of relatively dependable operation, I have been unable to download a single file via PremiumTo today - at most I get a 0 byte file. This is true for a variety of files located on uploadable, rapidgator, uptobox, uploaded.net and 1fichier. Other hosters I haven't tested, but the same probably applies.

Files download normally off PremiumTo's website.

I'd appreciate some panic.

**LATEST**
PremiumTo has switched to Cloudflare. It has been suggested this may have introduced an additional redirect, leading to the problem. For me, downloading eventually resumed by itself, without updates or modification of pyLoad, a week after I noticed the problem.

**Comment by Mister-XY** [comment]
Oh i have the same error 
WARNING	PremiumTo: Check result: Empty file | Waiting 1 minute and retry
Download Check triggered
Download Check triggered
PremiumTo: Handled as premium download
PremiumTo: Filesize: 0 KiB, Traffic left for user  12345: 123454567 KiB
PremiumTo: Account Info: {'validuntil': -1, 'premium': True, 'trafficleft': 123454567}
PremiumTo: Get Account Info for 12345
PremiumTo: File status: queued
PremiumTo: Current file info: {'status': 3, 'url': u'http://share-online.biz/dl/datei', 'name': u'datei', 'size': 0}
PremiumTo: Previous file info: {}
PremiumTo: Update file info...
UserAgentSwitcher: Use custom user-agent string: Mozilla/5.0 (Windows NT 6.1; WOW64; rv:37.0) Gecko/20100101 Firefox/37.0
Download starts: http://share-online.biz/dl/datei

**Comment by Mister-XY** [comment]
a new debug file
http://pastebin.com/vgCqp4Ph

**Comment by Mister-XY** [comment]
I have wrote with the support, they have build in an addional forwarding.

**Comment by stickell** [comment]
I'm asking the support about their changes to the APIs

**Comment by acid112** [comment]
i think i found a solution.
You need cookies to download files.
so i simply add two lines to this file in method "login" :
./userplugins/accounts/PremiumTo.py:

    def login(self, user, data, req):
        self.username = user
        self.password = data['password']
        authcode = req.load("http://premium.to/api/getauthcode.php",
                            get={'username': user, 'password': self.password},
                            decode=True)

        if "wrong username" in authcode:
            self.wrongPassword()
        #AddCookie
        cj = self.getAccountCookies(user)
        cj.setCookie("premium.to", "auth", authcode)

**Comment by acid112** [comment]
Now it is possible to download files for me.
Can someone verify this ?

**Comment by Mister-XY** [comment]
I get errors with pyload, if i start this with this script. But now i can download anyfiles but with wrong filenames.

**Comment by acid112** [comment]
yes, that is another problem...
If i add a link manually like http://uploaded.net/file/_______
everthing seems to be fine. Pyload get the filename and update it.
If i add a link like http://ul.to/_______ or uploaded.to/file/______
then pyload can not get the filename. I dont know why. 
In PluginThread.py pyload fetches the fileinformations.
If i debug the result-list (for example for ul.to/*), it seems ok. Pyload gets the correct filenames and filesizes, but there is no update....
There must be another error....

**Comment by stickell** [comment]
Strange because credentials are already passed to the API and the documentation doesn't say a cookie is also needed.
Let's wait for a reply by the support first, but if it fix the problem I will merge.

About the name issue pyLoad sets the name passed by premiumto through the content-disposition header. Maybe they don't set it for these hosts. However they must fix this issue, we can't do anything about.
However I can suggest a workaround: From the WebUI stop pyLoad before adding the links and restart only when you see in the log that the information retrieve is completed for the package added. This way the origin plugin (ex. UploadedTo) will have the time to retrieve all the infos before premiumto act.

**Comment by Mister-XY** [comment]
By me is not fixed.

**Comment by acid112** [comment]
ContentDisposition seems to be right:
12.04.2015 11:52:50	DEBUG	Content-Disposition: GT100SCN12012015.rar
1460	12.04.2015 11:52:58	INFO	Download finished: http://ul.to/ljrhXXXX
1461	12.04.2015 11:52:58	INFO	Package finished: aaaaassss
1462	12.04.2015 11:52:58	DEBUG	ExtractArchive: Use for extensions: 7z|.bz2|.bzip2|.gz|.gzip|.lha|.lzh|.lzma|.rar|.tar|.taz|.tbz|.tbz2|.tgz|.xar|.xz|.z|.zip
1463	12.04.2015 11:52:58	INFO	ExtractArchive: Check package: aaaaassss
1464	12.04.2015 11:52:59	DEBUG	All downloads processed
1465	12.04.2015 11:52:59	INFO	ExtractArchive: No files found to extract
1466	12.04.2015 11:52:59	DEBUG	All downloads finished 

But Filename in filesystem is: ./http:ul.toljrhXXXX
Pyload (ExtractArchive) cant find the File

And about the credentials:
There was a change. You can test this:
http://premium.to/api/getfile.php?username=YOUR USERNAME&password=YOUR PASSWORD&link=filesonic.com%2Ffile%2FHI7569
You can not download a file, until you are logged in (and the cookie is set)
If the cookie is set, you dont need the credentials

**Comment by acid112** [comment]
Here is a screenshot after adding the link to link container.
The first one is an ul.to-link.
The second one is the same FileID but in http://uploadet.net/file/ID

![zwischenablage-1](https://cloud.githubusercontent.com/assets/11908523/7105345/68d77442-e114-11e4-94c4-6884a52b924d.jpg)

**Comment by Mister-XY** [comment]
This is what i get

5744	12.04.2015 13:07:42	ERROR	Error importing PremiumTo: expected an indented block (PremiumTo.py, line 37)

**Comment by Mister-XY** [comment]
with the cookie script.

**Comment by acid112** [comment]
yes, a simple copy paste fault :)
or remove the hashtag-line
        if "wrong username" in authcode:
            self.wrongPassword()

        cj = self.getAccountCookies(user)
        cj.setCookie("premium.to", "auth", authcode)
        self.logDebug("Set AuthCookie for PremiumTo")

**Comment by Mister-XY** [comment]
12.04.2015 13:21:25	ERROR	Error importing PremiumTo: name 'authcode' is not defined
5996	12.04.2015 13:21:18	ERROR	Error importing PremiumTo: name 'authcode' is not defined

**Comment by acid112** [comment]
ok, please post the content of your accounts/PremiumTo.py

**Comment by Mister-XY** [comment]
# -*- coding: utf-8 -*-

from module.plugins.Account import Account


class PremiumTo(Account):
    __name__    = "PremiumTo"
    __type__    = "account"
    __version__ = "0.08"

    __description__ = """Premium.to account plugin"""
    __license__     = "GPLv3"
    __authors__     = [("RaNaN", "RaNaN@pyload.org"),
                       ("zoidberg", "zoidberg@mujmail.cz"),
                       ("stickell", "l.stickell@yahoo.it")]


    def loadAccountInfo(self, user, req):
        traffic = req.load("http://premium.to/api/straffic.php",
                           get={'username': self.username, 'password': self.password})

        if "wrong username" not in traffic:
            trafficleft = sum(map(float, traffic.split(';'))) / 1024  #@TODO: Remove `/ 1024` in 0.4.10
            return {'premium': True, 'trafficleft': trafficleft, 'validuntil': -1}
        else:
            return {'premium': False, 'trafficleft': None, 'validuntil': None}


    def login(self, user, data, req):
        self.username = user
        self.password = data['password']
        authcode = req.load("http://premium.to/api/getauthcode.php",
                            get={'username': user, 'password': self.password},
                            decode=True)

        if "wrong username" in authcode:
            self.wrongPassword()
		
		cj = self.getAccountCookies(user)
		cj.setCookie("premium.to", "auth", authcode)
		self.logDebug("Set AuthCookie for PremiumTo")

**Comment by acid112** [comment]
ok, there is no error.
Strange.
authcode is definied as you can see

**Comment by Mister-XY** [comment]
12.04.2015 13:29:09 ERROR     Error importing PremiumTo: unexpected indent (PremiumTo.py, line 39)
Traceback (most recent call last):
  File "/usr/bin/pyLoadCore", line 667, in <module>
    main()
  File "/usr/bin/pyLoadCore", line 658, in main
    pyload_core.start()
  File "/usr/bin/pyLoadCore", line 433, in start
    self.accountManager.getAccountInfos()
  File "/usr/share/pyload/module/utils.py", line 165, in new
    return func(*args)
  File "/usr/share/pyload/module/plugins/AccountManager.py", line 175, in getAccountInfos
    p = self.getAccountPlugin(p)
  File "/usr/share/pyload/module/plugins/AccountManager.py", line 55, in getAccountPlugin
    self.plugins[plugin] = self.core.pluginManager.loadClass("accounts", plugin)(self, self.accounts[plugin])
TypeError: 'NoneType' object is not callable

**Comment by acid112** [comment]
unexpected indent 
it is a syntactic error.
maybe there are spaces or tabs before the new lines.
Check, if there are spaces from the left of the new lines, remove the empty line, count the spaces. The count of the spaces from the left from the new lines  must be the same as the if-line.
Check it. 
It is no error, it is just a copy paste mistake

**Comment by Mister-XY** [comment]
Can you upload your file to http://pastebin.com/

**Comment by acid112** [comment]
i dont know how to upload files on paste.
http://www.file-upload.net/download-10535109/PremiumTo.py.html
Click on the right blue DownloadButton

**Comment by Mister-XY** [comment]
Ok, i test it.

**Comment by sasese** [comment]
@stickell Any news?

**Comment by stickell** [comment]
Support told me they made no changes to the APIs.
However they moved to CloudFlare so there is an extra redirect

Sorry guys I'm very very busy so I have very little time to work on this

**Comment by Mister-XY** [comment]
This is not good.

**Comment by sasese** [comment]
perhaps @vuolter can look into this.

**Comment by sasese** [comment]
For me PremiumTo is working again, without any modification of pyLoad.

**Comment by Mister-XY** [comment]
Wich hoster do you use in Premium.to?

**Comment by sasese** [comment]
So far, I have noticed uploaded.net and rapidgator back to normal, possibly uptobox and netload, too.

**Comment by sasese** [comment]
**CONCLUSION**
PremiumTo has switched to Cloudflare. It has been suggested this may have introduced an additional redirect, leading to the problem. For me, downloading eventually resumed by itself, without updates or modification of pyLoad, a week after I noticed the problem

In the absence of further activity in this thread for eight days, I'm closing the issue on the presumption it has been resolved.
